pkg/auth/creds_config.go

```diff
@@ -0,0 +1,136 @@
+
+// loadConfig loads the configuration
+// from the config file
+func loadConfig() (*ConfigFile, error) {
+	config, _ := loadNotationConfig()
+	if config != nil && !config.isEmptyConfig() {
+		fmt.Println("Using notation config file")
+		return config, nil
+	}
+
+	config, _ = loadDockerConfig()
+	if config != nil && !config.isEmptyConfig() {
+		fmt.Println("Using docker config file")
+		return config, nil
+	}
+	if !config.ContainsAuth() {
+		fmt.Println("Credentials store config is not set up, using the default setting")
+		config.CredentialsStore = DetectDefaultStore(config.CredentialsStore)
+	}
+	return config, nil
+}
+
+// loadDockerConfig loads the configuration
+// from the config file under .notation directory
+func loadNotationConfig() (*ConfigFile, error) {
+	notationConfig, err := config.LoadOrDefault()
+	if err != nil {
+		return nil, err
+	}
+	config := NewConfigFromNotationConfig(notationConfig)
+	return config, nil
+}
+
+// loadDockerConfig loads the configuration
+// from the config file under .docker directory
+func loadDockerConfig() (*ConfigFile, error) {
+	dockerConfig, err := dockerConfig.Load(dockerConfig.Dir())
+	if err != nil {
+		return nil, err
+	}
+	config := NewConfigFromDockerConfig(dockerConfig)
+	return config, nil
+}
+
+// isEmptyConfig checks if the config is empty
+func (configFile *ConfigFile) isEmptyConfig() bool {
+	return configFile.CredentialsStore == "" && len(configFile.CredentialHelpers) == 0
+}
+
+// GetCredentialsStore returns a new credentials store from the settings in the
+// configuration file
+func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) CredentialStore {
+	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
+		return newNativeStore(helper)
+	}
+	return nil
+}
+
+// var for unit testing.
+var newNativeStore = func(helperSuffix string) CredentialStore {
+	return NewNativeAuthStore(helperSuffix)
+}
+
+// getConfiguredCredentialStore returns the credential helper configured for the
+// given registry, the default credsStore, or the empty string if neither are
+// configured.
+func getConfiguredCredentialStore(c *ConfigFile, registryHostname string) string {
+	if c.CredentialHelpers != nil && registryHostname != "" {
+		if helper, exists := c.CredentialHelpers[registryHostname]; exists {
+			return helper
+		}
+	}
+	return c.CredentialsStore
+}
+
+// DetectDefaultStore return the default credentials store for the platform if
+// the store executable is available.
+func DetectDefaultStore(store string) string {
+	platformDefault := defaultCredentialsStore()
+
+	// user defined or no default for platform
+	if store != "" || platformDefault == "" {
+		return store
+	}
+
+	if _, err := exec.LookPath(remoteCredentialsPrefix + platformDefault); err == nil {
+		return platformDefault
+	}
+	return ""
+}
+
+func defaultCredentialsStore() string {
+	if _, err := exec.LookPath("pass"); err == nil {
+		return "pass"
+	}
+
+	return "secretservice"
+}
```

Comment:
It seems not cross-platform capable. Let's not detect default store for this PR.

Reply:
Gotcha, I'll add TODO comment.